docs/adrs/006-web-apis.md

```diff
@@ -39,9 +39,21 @@ that manages the injected connections and provides methods for interacting with
 Creating the `client` should be the starting point for any application working with Penumbra:
 
 ```ts
-import { createPenumbraClient } from '@penumbra-zone/client';
+import { PenumbraClient } from '@penumbra-zone/client';
 
-export const client = createPenumbraClient();
+const providerManifests: Record<
+  string,
+  Promise<PenumbraManifest>
+> = PenumbraClient.providerManifests();
+const someProviderOrigin: keyof providerManifests = '....';
+
+// connect will fetch and verify manifest before initiating connection, then return an active client
+const someProviderClient = await PenumbraClient.connect(someProviderOrigin);
```

Comment:
issue: the `providerClient` has to be reused across the users' apps. If we require them to know the origin beforehand, idk how they are going to reuse it...

suggestion: allow the creation of a client without the need of the origin. Seriously, why changing this?

Reply:
specifying an origin is selecting a provider.

there is no way to get around selecting a provider deliberately.

selection of a provider is creation of state, and the object instance is returned to track that state and represent the client connection. the consumer is expected to handle the client instance and convey it to where it should be available.

if consumers want to recreate clients to the same provider, that should be supported, but then they have to be responsible for their own state tracking.

in react apps this can be provided by hooks in a react package, but all of that pends merge of a settled client interface.

Reply:
notably if this is not changed, the API previously defined in this doc provides no way for a page to inspect a provider's connection state before attempting to connect, and given the requirement that client creation not automatically attempt to connect, then that results in a user experience of initiating every connection manually.